applications/OptimizationApplication/python_scripts/execution_policies/kratos_analysis_execution_policy.py

```diff
@@ -0,0 +1,112 @@
+from importlib import import_module
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.analysis_stage import AnalysisStage
+from KratosMultiphysics.OptimizationApplication.execution_policies.execution_policy import ExecutionPolicy
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import GetClassModuleFromKratos
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem) -> ExecutionPolicy:
+    if not parameters.Has("name"):
+        raise RuntimeError(f"KratosAnalysisExecutionPolicy instantiation requires a \"name\" in parameters [ parameters = {parameters}].")
+    if not parameters.Has("settings"):
+        raise RuntimeError(f"KratosAnalysisExecutionPolicy instantiation requires a \"settings\" in parameters [ parameters = {parameters}].")
+
+    return KratosAnalysisExecutionPolicy(parameters["name"].GetString(), model, parameters["settings"], optimization_problem)
+
+class KratosAnalysisExecutionPolicy(ExecutionPolicy):
+    def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__(name)
+
+        default_settings = Kratos.Parameters("""{
+            "model_part_names" : [],
+            "analysis_module"  : "KratosMultiphysics",
+            "analysis_type"    : "",
+            "analysis_settings": {}
+        }""")
+        self.model = model
+        self.parameters = parameters
+        self.optimization_problem = optimization_problem
+        self.parameters.ValidateAndAssignDefaults(default_settings)
+
+        analysis_module = parameters["analysis_module"].GetString()
+        analysis_type = parameters["analysis_type"].GetString()
+        analysis_settings = parameters["analysis_settings"]
+
+        if analysis_module == "KratosMultiphysics":
+            analysis_module = GetClassModuleFromKratos(analysis_type)
+
+        self.model_parts = []
+        analysis_full_module = f"{analysis_module}.{Kratos.StringUtilities.ConvertCamelCaseToSnakeCase(analysis_type)}"
+        self.analysis: AnalysisStage = getattr(import_module(analysis_full_module), analysis_type)(self.model, analysis_settings.Clone())
+
+        if analysis_settings["output_processes"].Has("vtk_output"):
+           self._GetOutputData(analysis_settings["output_processes"]["vtk_output"][0]["Parameters"])
+        elif analysis_settings["output_processes"].Has("gid_output"):
+           self._GetOutputData(analysis_settings["output_processes"]["gid_output"][0]["Parameters"])
```

Comment:
This wil break in HDF5, VTU outputs as well as if user specifies arbitrary namings.